spec/board_spec.rb

```diff
@@ -15,4 +15,28 @@
       # require 'pry';binding.pry
     end
   end
+
+  describe "#board_is_full" do 
+    it "returns true if the first row is full" do 
+      player = Player.new
+      board = Board.new
+      turn = Turn.new(player, board)
+
+      
+      expect(board.board_is_full?).to eq(false)
+      
+      board.board_grid[0][0].set_state("x")
+      board.board_grid[0][1].set_state("x")
+      board.board_grid[0][2].set_state("x")
+      board.board_grid[0][3].set_state("x")
+      board.board_grid[0][4].set_state("x")
+      board.board_grid[0][5].set_state("x")
```

Comment:
So we did this manually on purpose to avoid having any logic/iteration in tests - idea being tests should be very clear and deliberate to prove something.